core-bundle/assets/controllers/main-menu-controller.js

```diff
@@ -0,0 +1,25 @@
+import { Controller } from '@hotwired/stimulus';
+import DropdownController from './dropdown-controller';
+
+export default class extends DropdownController {
+    static values = {
+        bodyClass: {
+            type: String,
+            default: 'show-navigation',
+        },
+    };
+
+    toggle() {
+        const isOpen = this.buttonTarget.ariaExpanded === 'true';
+
+        document.body.classList.toggle(this.bodyClassValue, !isOpen);
```

Comment:
any particular reason this cannot be part of the `DropdownController`, and check for `hasBodyClassValue` before applying it? Also, why using `classList.toggle` and not `add/remove`?

Reply:
Yes. The controller is applied on the body in this case (due to the target being `#left`) and thus would have many targets, be it the profile, the split button and other future controllers.

Reply:
it wouldn't be a problem here due to https://stimulus.hotwired.dev/reference/controllers#nested-scopes, but it would not be possible to have another global scope (which is the case with a separate controller as well though).

Reply:
Didn't see that one in the docs. Would you prefer doing it this way then?

Reply:
I just realized that we don't even need the bodyClassValue as the controller is already on the body and we could simply add a class as it used to within the profile-controller 😁 gonna simplify it.

Reply:
And to that:
> Also, why using classList.toggle and not add/remove?

The second parameter of classList.toggle allows adding or removing (based on true or false), so:

```js
if (this.buttonTarget.ariaExpanded === 'true') {
    document.body.classList.remove(this.bodyClassValue);
} else {
    document.body.classList.add(this.bodyClassValue);
}
```

could also be simplified to a oneliner
```js
document.body.classList.remove(this.bodyClassValue, this.buttonTarget.ariaExpanded === 'false');
```